src/parser/md.rs

```diff
@@ -558,8 +558,8 @@ impl MdParser {
     ) -> Vec<(String, String)> {
         let mut entries = vec![];
         let pref = &self.file_prefix;
-        for block in doc.code_blocks(language) {
-            if let Some(name) = &block.name {
+        for (name, _block) in doc.code_blocks_by_name(language) {
```

Comment:
I think it would be a better change to make `entries` a `HashSet<(&str, &str)>` here instead of building the intermediate `HashMap` that is basically discarded.

Reply:
I.e. add another function for only getting the names? Because the map is required in printing.

Reply:
I would just change the return type of `get_entry_points` to `HashSet<(&str, &str)>` (or maybe the owned variant of that). (This cannot depend on order as `code_blocks_by_name` breaks order anyway.)

Reply:
Sure, but this means returning a HashSet instead of a Vector, but your original comment was about to avoid the HashMap. Which I thought referred to the return type of `code_blocks_by_name`, as there is no other HashMap involved.

Reply:
> but your original comment was about to avoid the HashMap.

I meant to avoid the `HashMap` returned by `code_blocks_by_name` which is thrown away and instead using a `HashSet` that stays around replacing the `Vec` that stays around. `get_entry_points` does not need to call `code_blocks_by_name` at all just to collapse duplicate names, it can do that itself, e.g.
```rust
pub fn get_entry_points<'a>(
    &self,
    doc: &'a Document,
    language: Option<&'a str>,
) -> HashMap<&'a str, &'a str> {
    let mut entries = HashMap::new();
    let pref = &self.file_prefix;
    for block in doc.code_blocks(language) {
        if let Some(name) = &block.name {
            if let Some(rest) = name.strip_prefix(pref) {
                entries.insert(name.as_str(), rest);
            }
        }
    }
    entries
}
```

Reply:
(This this does not even seem to require any changes at the call sites.)

Reply:
Ah, now I got it. Ignore my the commit I just made ;)

Reply:
It is not 100% equivalent, as the check for dublicates is now done after stripping the prefix. Not sure if this could be a problem. Only when there are equally-named code blocks, but one with `file:` and one without. In theory, that would be allowed.

Reply:
> It is not 100% equivalent, as the check for dublicates is now done after stripping the prefix. Not sure if this could be a problem.

At least in the code above, the full name is used as the key so duplicates are checked exactly as before.

Reply:
But on the other hand, it seems blocks without file prefix don't even arrive here. Not sure but I guess an unnamed entrypoint already erts renamed to `file:<target>` already before.

Reply:
> Not sure but I guess an unnamed entrypoint gets renamed to file:<target> already before.

Isn't this
```rust
    let mut entries = vec![(entrypoint.as_deref(), file_name.to_owned())];
    let extra_entries = parser.get_entry_points(&document, language);

    entries.extend(
        extra_entries
            .iter()
            .map(|(e, p)| (Some(&e[..]), PathBuf::from((*p).to_owned() + ".temp"))),
    );
```
?

Reply:
It does...

Reply:
> Ah, now I got it. Ignore my the commit I just made ;)

This refers to the commit to collects into a `HashSet` and then into a `Vec`?

Reply:
Yes. I think this is resolved not with my force-push.